src/v/cloud_storage_clients/s3_client.cc

```diff
@@ -401,7 +404,13 @@ ss::future<ResultT> parse_head_error_response(
             code = fmt::format("{}", status_to_error_code(hdr.result()));
             msg = ss::sstring(hdr.reason().data(), hdr.reason().size());
         }
-        auto rid = hdr.at(aws_header_names::x_amz_request_id);
+        std::string_view rid;
+        if (hdr.find(aws_header_names::x_amz_request_id) != hdr.end()) {
+            rid = hdr.at(aws_header_names::x_amz_request_id);
+        } else if (
+          hdr.find(aws_header_names::x_guploader_uploadid) != hdr.end()) {
+            rid = hdr.at(aws_header_names::x_guploader_uploadid);
```

Comment:
Does azure need it's own header?

Reply:
Azure has its own client at https://github.com/redpanda-data/redpanda/blob/9e0e93a61ef08ae4f01927463fb5c007419f35b0/src/v/cloud_storage_clients/abs_client.cc#L1